### PR TITLE
update wifi ap enable to use get opt parsing

### PR DESCRIPTION
**Pull request by raintamer**
The commit 8256d02d3a9625c03e1d584cf54b1f0cc1ae164b introduced getopt parsing for the wifi connect command This needs to be updated for the wifi ap enable since ap enable also uses the same parsing as connect.

### CODE REVIEW

File: subsys/net/l2/wifi/wifi_shell.c [hunk 1]
```diff
@@ -496,12 +496,8 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 			}
 			break;
 		case 'p':
-			if (secure_connection) {
```
Comment (rlubos) [inline subsys/net/l2/wifi/wifi_shell.c:499]: Is this change related to option ordering? This should be reflected in the commit message, or even better put in a separate commit with the fix description in the message.

File: subsys/net/l2/wifi/wifi_shell.c [hunk 2]
```diff
@@ -568,15 +564,17 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 				&params->bssid[4], &params->bssid[5]);
 			break;
 		case 'h':
-			shell_help(sh);
+			return -ENOEXEC;
```
Comment (rlubos) [inline subsys/net/l2/wifi/wifi_shell.c:567]: Why this change?

**Review by rlubos — COMMENTED** [review]
Please use `net: wifi: shell:` prefix in the commit title. Please check the compliance results, the commit message lines are too long.

File: subsys/net/l2/wifi/wifi_shell.c (hunk 1 above)
Comment (raintamer) [reply subsys/net/l2/wifi/wifi_shell.c:499]: There is no longer a requirement for ordered options with this commit. The previous example would fail to set the PSK but allow the ap enable to proceed on mis ordered options. It now only warns if the PSK is set without the proper security. getopts allows us to not require an order to the options.

File: subsys/net/l2/wifi/wifi_shell.c (hunk 2 above)
Comment (raintamer) [reply subsys/net/l2/wifi/wifi_shell.c:567]: Without this change, wifi ap  enable -h will print the help then configure an AP with the default SSID an no security

File: subsys/net/l2/wifi/wifi_shell.c (hunk 1 above)
Comment (raintamer) [reply subsys/net/l2/wifi/wifi_shell.c:499]: Should a configuration that has a PSK but does not specify the type of security succeed or fail?

**Review by rlubos — APPROVED** [review]

File: subsys/net/l2/wifi/wifi_shell.c [hunk 3]
```diff
@@ -568,15 +564,17 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
...
 			break;
 		default:
 			PR_ERROR("Invalid option %c\n", opt);
 			shell_help(sh);
 			return -EINVAL;
 		}
 	}
-
+	if (params->psk && !secure_connection) {
+		PR_WARNING("Passphrase provided without security configuration\n");
```
Comment (rado17) [inline subsys/net/l2/wifi/wifi_shell.c:576]: Should we return error (`-EINVAL`) here or may be specify that since `security` is not configured, setting will default to `NONE` and passphrase will not be used?

Comment (raintamer) [reply subsys/net/l2/wifi/wifi_shell.c:576]: No when WIFI 6 is added it will use the saepasspharse instead of the passpharase so it would be valid not to have passphrase configured. Also although I cannot think of a reason the user may want a null passphrase. A warning should be sufficient.

File: subsys/net/l2/wifi/wifi_shell.c (hunk 1 above)
Comment (krish2718) [reply subsys/net/l2/wifi/wifi_shell.c:499]: FYI, we already do support setting either password (0-63) or SAE passphrase (no length limits). But this isn't exposed to user, it would still use the same `-p` arg, just the validation varies as per `-k`, so, can probably fail here.

**Review by jukkar — APPROVED** [review]
